uboot-cve-2019-13103-13106/README.md

```diff
@@ -0,0 +1,63 @@
+# uboot CVE 2019-13103 to 2019-13106
+
+This repo replicates finding [CVE-2019-13103](https://nvd.nist.gov/vuln/detail/CVE-2019-13103) and following CVE's. Many crashes are found including floating point errors and stack corruption bugs (Max [CVSS Score](https://nvd.nist.gov/vuln-metrics/cvss): 8.3).
+
+Das U-Boot is an open-source bootloader for embedded devices. Usually, it is
+the first thing the CPU runs, and it will boot into some form of linux from
+storage media. To fuzz it, we use the sandbox feature of u-boot, which
+compiles it as a linux userspace program so that features can be tested off
+of the embedded device. We stress the ext4 code by listing files.
+
+## To build
+
+Assuming you just want to build the docker image, run:
+
+```bash
+docker build -t forallsecure/uboot-cve-2019-13103-13106 .
+```
+
+## Get from Dockerhub
+
+If you don't want to build locally, you can pull a pre-built image
+directly from dockerhub:
+
+```bash
+docker pull forallsecure/uboot-cve-2019-13103-13106
+```
+
+
+## Run under Mayhem
+
+Change to the `uboot-cve-2019-13103-13106` folder and run:
+
+```bash
+mayhem run mayhem/uboot
+```
+
+and watch Mayhem replicate the bug! This bug takes a little time (~1 hour)
+to find but is still very quick!
+
+## Run locally
+
+To debug crashes locally, you need to be inside one of the previously built docker containers.
+
+Once inside the docker run:
+
+```bash
+cd /workdir
+/mayhem/u-boot-2019.07-rc4/u-boot -c "host bind 0 /PATH/TO/CRASH ; ls host 0"
+And replace the path to crash with any crash in the crash file.
+```
+
+## POC
```

Comment:
Can you add a new section that mentions we had to change AFL for it to repro? That caught me by surprise looking through the diff, and I could imagine someone missing that.

Reply:
The change is actually to the uboot code base for deferred fork server with afl fast clang

Reply:
(will still add note in readme though)